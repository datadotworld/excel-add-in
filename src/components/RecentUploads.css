/*
 * Copyright 2018 data.world, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This product includes software developed at
 * data.world, Inc. (http://data.world/).
 */

.item {
  background-color: #fff;
  position: relative;
  border-bottom: 1px solid #dfe3e9;
  padding: 10px 16px;
  display: flex;
  align-items: center;
  width: 100%;
  cursor: pointer;
}

.recents-item-text {
  width: 100%;
  overflow: hidden;
}

.recent-uploads-filename {
  color: #4e5057;
  font-size: 14px;
  font-weight: bold;
}

.recent-uploads-dataset {
  color: #4e5057;
  font-size: 14px;
}

.recent-uploads-sheet {
  color: #8c9caf;
  font-size: 14px;
}

.actions-container {
  display: flex;
}

.icon-upload-blue {
  width: 18px;
  height: 18px;
}

.icon-clear {
  color: #8c9caf;
  margin-right: 1rem;
}

.clear-warning-modal {
  padding-top: 50%;
}