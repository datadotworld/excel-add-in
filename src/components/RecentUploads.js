import React, { Component } from 'react';

import {
  Button,
  ControlLabel,
  Image,
  Tooltip,
  OverlayTrigger
} from 'react-bootstrap';
import { getDisplayRange } from '../util';
import './RecentUploads.css';

const upload = require('./icons/icon-upload-blue.svg');

class RecentItem extends Component {
  state = {
    loading: false,
    sheetName: ''
  };

  sync = async (filename, rangeAddress, dataset, worksheetId) => {
    this.setState({ loading: true });

    try {
      await this.props.sync(filename, rangeAddress, dataset, worksheetId);
      this.setState({ loading: false });
    } catch (syncError) {
      this.props.setError(syncError);
    }
  };

  componentDidMount() {
    this.props.getSheetName(this.props.worksheetId).then((sheetName) => {
      this.setState({ sheetName });
    });
  }

  render() {
    const { filename, dataset, rangeAddress, worksheetId } = this.props;

    const datasetLocation = `${dataset.owner}/${dataset.id}`;

    const rangeToShow = getDisplayRange(rangeAddress, this.state.sheetName);
    return (
      <div>
        <div className="item recent">
          <div className="recents-item-text">
            <div>
              <div className="recent-uploads-filename">{filename}</div>
            </div>
            <a
              href={`https://data.world/${datasetLocation}/workspace/file?filename=${filename}`}
              target="_blank"
              className="link button-link"
            >
              <div className="recent-uploads-dataset">{datasetLocation}</div>
            </a>
            <div className="info">{`Upload from: ${rangeToShow}`}</div>
          </div>
          <div
            className="icon-border"
            onClick={() =>
              this.sync(filename, rangeAddress, dataset, worksheetId)
            }
          >
            <div className="resync-icon">
              {this.state.loading ? (
                <div className="loader-icon" />
              ) : (
                <OverlayTrigger
                  placement="bottom"
                  overlay={<Tooltip id="tooltip">Upload</Tooltip>}
                >
                  <Image className="icon-upload-blue" src={upload} />
                </OverlayTrigger>
              )}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default class RecentUploads extends Component {
  render() {
    const { forceShowUpload, matchedFiles, getSheetName } = this.props;

    let previousDate = '';
    let showDate;

    return (
      <div>
        <div className="full-screen-modal category-title">
          <ControlLabel className="large-header">Recent Uploads</ControlLabel>
          <Button
            bsStyle="link"
            className="upload-button"
            onClick={() => forceShowUpload()}
          >
            + New upload{' '}
          </Button>
        </div>
        {matchedFiles.map((file) => {
          const dateArray = new Date(file.date).toDateString().split(' ');
          let dateToShow = dateArray[1] + ' ' + dateArray[2];
          if (dateToShow !== previousDate) {
            previousDate = dateToShow;
            showDate = true;
          } else {
            showDate = false;
          }

          return (
            <div key={file.filename}>
              {showDate && <div className="date">{dateToShow}</div>}
              <RecentItem
                {...file}
                sync={this.props.sync}
                setError={this.props.setError}
                getSheetName={getSheetName}
              />
            </div>
          );
        })}
      </div>
    );
  }
}
