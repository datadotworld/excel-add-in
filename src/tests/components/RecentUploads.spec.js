import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import RecentUploads from '../../components/RecentUploads';

const localStorageMock = {
    getItem: jest.fn()
    };
global.localStorage = localStorageMock;

it('renders modal', () => {
  expect(renderer.create(<RecentUploads matchedFiles={[]}/>).toJSON()).toMatchSnapshot()
});