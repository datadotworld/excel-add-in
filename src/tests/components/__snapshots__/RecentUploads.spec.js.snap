// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders modal 1`] = `
<div>
  <div
    className="full-screen-modal category-title"
  >
    <label
      className="large-header control-label"
      htmlFor={undefined}
    >
      Recent Uploads
    </label>
    <button
      className="upload-button btn btn-link"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      + New upload 
    </button>
  </div>
  <div
    className="category-reminder"
  >
    Showing 
    0
     most recent uploads
  </div>
</div>
`;
